### ISSUE TITLE
Django commit c049c43 landing in 1.9 breaks TaggableManager

### ISSUE THREAD

**Issue by nealtodd** [alex/django-taggit#360]
The https://github.com/django/django/commit/c049c43e28a3e46eaed13bb8e018b221250a0145 commit on Django 1.9.x stable (backported from master) and part of 1.9rc1 breaks TaggableManager during system checks, preventing an application from starting up:

```
Performing system checks...

Unhandled exception in thread started by <function wrapper at 0xb569579c>
Traceback (most recent call last):
  File "/django/utils/autoreload.py", line 226, in wrapper
    fn(*args, **kwargs)
  File "/django/core/management/commands/runserver.py", line 116, in inner_run
    self.check(display_num_errors=True)
  File "/django/core/management/base.py", line 426, in check
    include_deployment_checks=include_deployment_checks,
  File "/django/core/checks/registry.py", line 75, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/django/core/checks/model_checks.py", line 28, in check_all_models
    errors.extend(model.check(**kwargs))
  File "/django/db/models/base.py", line 1170, in check
    errors.extend(cls._check_fields(**kwargs))
  File "/django/db/models/base.py", line 1249, in _check_fields
    errors.extend(field.check(from_model=cls, **kwargs))
  File "/django/db/models/fields/related.py", line 118, in check
    errors.extend(self._check_relation_model_exists())
  File "/django/db/models/fields/related.py", line 152, in _check_relation_model_exists
    rel_is_missing = self.remote_field.model not in self.opts.apps.get_models()
AttributeError: 'TaggableManager' object has no attribute 'opts'
```

I haven't figured out yet where `self.opts` gets set on a `Field` or `RelatedField` class (which `TaggableManager` subclasses) during app config. ~~I presume that because `TaggableManager` is slightly unusual in being a `django.db.models.Manager` as well as subclassing `RelatedField` and `Field` it might not be getting `self.opts` set in the way a model field would.~~ (edit: misread code, it's not a Manager, it's a field using _TaggableManager)

(Aside: why does `TaggableManager` subclass `RelatedField` and `Field` when `RelatedField` itself subclasses  `Field`?)

I'll do some more digging but I thought I'd report the issue now ahead of the scheduled Django 1.9 release in a few days (1st Dec).

**Comment by collinanderson** [comment]
I ran into this yesterday.

**Comment by frewsxcv** [comment]
Ping me if you find a fix and I'll happily merge it

**Comment by collinanderson** [comment]
Looks like it needs to get set in `contribute_to_class`:
https://github.com/django/django/blob/c049c43e28a3e46eaed13bb8e018b221250a0145/django/db/models/fields/related.py#L295

**Comment by collinanderson** [comment]
Yes, adding `self.opts = cls._meta` in `contribute_to_class` seems to fix it for me. See PR #361.

**Comment by nealtodd** [comment]
Ah yes, I hadn't noticed the `contribute_to_class` on `TaggableManager` itself. As Fields and Managers are normally added to a model class (rather than being subclasses here) I was scratching my head over how it'd be possible from `RelatedField.contribute_to_class`! Thanks @collinanderson, that PR should do the trick.

**Comment by lineway** [comment]
I use django==1.9 and django-taggit==0.17.2 to build a project ,when I want to migrate,it's happen this error
`Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/zhangyiming/anaconda/envs/webdev3/lib/python3.5/site-packages/django/core/management/__init__.py", line 350, in execute_from_command_line
    utility.execute()
  File "/Users/zhangyiming/anaconda/envs/webdev3/lib/python3.5/site-packages/django/core/management/__init__.py", line 342, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/zhangyiming/anaconda/envs/webdev3/lib/python3.5/site-packages/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/zhangyiming/anaconda/envs/webdev3/lib/python3.5/site-packages/django/core/management/base.py", line 398, in execute
    self.check()
  File "/Users/zhangyiming/anaconda/envs/webdev3/lib/python3.5/site-packages/django/core/management/base.py", line 426, in check
    include_deployment_checks=include_deployment_checks,
  File "/Users/zhangyiming/anaconda/envs/webdev3/lib/python3.5/site-packages/django/core/checks/registry.py", line 75, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/Users/zhangyiming/anaconda/envs/webdev3/lib/python3.5/site-packages/django/core/checks/model_checks.py", line 28, in check_all_models
    errors.extend(model.check(**kwargs))
  File "/Users/zhangyiming/anaconda/envs/webdev3/lib/python3.5/site-packages/django/db/models/base.py", line 1170, in check
    errors.extend(cls._check_fields(**kwargs))
  File "/Users/zhangyiming/anaconda/envs/webdev3/lib/python3.5/site-packages/django/db/models/base.py", line 1249, in _check_fields
    errors.extend(field.check(from_model=cls, **kwargs))
  File "/Users/zhangyiming/anaconda/envs/webdev3/lib/python3.5/site-packages/django/db/models/fields/related.py", line 118, in check
    errors.extend(self._check_relation_model_exists())
  File "/Users/zhangyiming/anaconda/envs/webdev3/lib/python3.5/site-packages/django/db/models/fields/related.py", line 152, in _check_relation_model_exists
    rel_is_missing = self.remote_field.model not in self.opts.apps.get_models()
AttributeError: 'TaggableManager' object has no attribute 'opts'`
How can I solute this error?

**Comment by nealtodd** [comment]
@lineway upgrade your django-taggit to the current version (0.21.6), this wasn't fixed until the 0.17.5 release (see CHANGELOG.txt).
